### PR TITLE
Set readable names for date objects.

### PR DESCRIPTION
**Pull request by zinoviev**
PR for https://github.com/facebook/react-devtools/issues/388

### CODE REVIEW

**Comment by gaearon** [comment]
If you look at https://github.com/facebook/react-devtools/issues/388, you will see that there is a discussion with another person working on it, and a label saying "someone is working on this". I appreciate the pull request, but I will close this without reviewing because we need to give the person who already started the work a chance to finish it. If he gets stuck, and we can't work it out, or loses interest, I'll remove the "someone is working on this" label, and the task will be up for grabs again. Thanks!

**Comment by zinoviev** [comment]
Sory folks, I started work on this after twitter 'call to arms' from @gaearon and this issue wasn't assigned at moment. Really not intended to race someone. Now I understand I should write here on the beginning, and next time I will checkin on github before PR. 

I'm not expert how people usually do things in open source, but maybe it is worth to write about "someone is working on this" workflow in contributing.md?

**Comment by gaearon** [comment]
Reopening since #559 received no followup.
Could you please rebase this?

File: agent/dehydrate.js [hunk 1]
```diff
@@ -77,8 +77,10 @@ function dehydrate(data: Object, cleaned: Array<Array<string>>, path?: Array<str
   // TODO when this is in the iframe window, we can just use Object
   if (data.constructor && typeof data.constructor === 'function' && data.constructor.name !== 'Object') {
     cleaned.push(path);
+    const isDate = data.constructor.name === 'Date';
```
Comment (gaearon) [inline agent/dehydrate.js]: Is this check guaranteed to work? Maybe we can do something safer, like `Object.prototype.toString.call(data)`?

**Comment by zinoviev** [comment]
@gaearon rebased and updated

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**